client-js/src/client/backend-client.js

```diff
@@ -45,16 +47,34 @@ export class BackendClient {
     }
 
     /**
-     * Fetches all the entities of the given type.
-     *
-     * @param type          the target {@link TypeUrl}
-     * @param dataCallback  the callback which receives the data, one-by-one, in
-     *                      a form of a JS object
-     * @param errorCallback the callback which receives the errors
+     * Defines a fetch query of all objects of specified type.
+     * 
+     * @param ofType {TypeUrl} a type of the entities to be queried
+     * @returns {{oneByOne, atOnce}} an object allowing two fetch strategies: one-by-one or all-at-once.
+     * @example 
+     * // Fetch items one-by-one using an Observable.
+     * // Suitable for big collections.
+     * fetchAll({ofType: taskType}).oneByOne().subscribe({
+     *   next(value) { ... },
+     *   error(error) { ... },
+     *   complete() { ... }
+     * })
+     * @example 
+     * // Fetch all items at once using a Promise.
+     * fetchAll({ofType: taskType}).atOnce().then(data => { ... })
      */
-    fetchAll(type, dataCallback, errorCallback = null) {
-        let query = this._actorRequestFactory.queryAll(type);
-        this._fetch(query, dataCallback, errorCallback);
+    fetchAll({ofType: typeUrl}) {
+        let query = this._actorRequestFactory.queryAll(typeUrl);
```

Comment:
Not your code originally, but, while we are here, let's replace this with `const` since it's not reassigned (I think you already follow the same convention in your changes).